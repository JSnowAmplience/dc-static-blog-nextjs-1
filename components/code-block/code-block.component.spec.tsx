import renderer from 'react-test-renderer';
import CodeBlock from './code-block.component';

describe('CodeBlock', (): void => {
  it('should render some code as a pre tag with syntax highlighting', (): void => {
    const code: { language: string; value: string } = {
      language: 'javascript',
      value: "```\nimport { something } from 'somewhere';\n\nconst someThing = something();\n```"
    };

    const wrapper = renderer.create(<CodeBlock language={code.language} value={code.value} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
