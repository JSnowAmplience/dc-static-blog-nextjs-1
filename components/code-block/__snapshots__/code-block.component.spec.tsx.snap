// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock should render some code as a pre tag with syntax highlighting 1`] = `
<pre
  style={
    Object {
      "backgroundColor": "#fff",
    }
  }
>
  <code
    style={
      Object {
        "float": "left",
        "paddingRight": "10px",
      }
    }
  >
    <span
      className="react-syntax-highlighter-line-number"
      style={Object {}}
    >
      1

    </span>
    <span
      className="react-syntax-highlighter-line-number"
      style={Object {}}
    >
      2

    </span>
    <span
      className="react-syntax-highlighter-line-number"
      style={Object {}}
    >
      3

    </span>
    <span
      className="react-syntax-highlighter-line-number"
      style={Object {}}
    >
      4

    </span>
    <span
      className="react-syntax-highlighter-line-number"
      style={Object {}}
    >
      5

    </span>
  </code>
  <code>
    <span
      className="token template-string token string token template-string token string token template-string token string"
      style={Object {}}
    >
      <span
        className="token template-string token template-punctuation string"
        style={Object {}}
      >
        \`
      </span>
      <span
        className="token template-string token template-punctuation string"
        style={Object {}}
      >
        \`
      </span>
      <span
        className="token template-string token template-punctuation string"
        style={Object {}}
      >
        \`
      </span>
      <span
        className="token template-string token string"
        style={Object {}}
      >
        

      </span>
    </span>
    <span
      className="token template-string token string token template-string token string token template-string token string"
      style={Object {}}
    >
      import { something } from 'somewhere';

    </span>
    <span
      className="token template-string token string token template-string token string token template-string token string"
      style={Object {}}
    >
      

    </span>
    <span
      className="token template-string token string token template-string token string token template-string token string"
      style={Object {}}
    >
      const someThing = something();

    </span>
    <span
      className="token template-string token string token template-string token string token template-string token string"
      style={Object {}}
    >
      <span
        className="token template-string token string"
        style={Object {}}
      >
        
      </span>
      <span
        className="token template-string token template-punctuation string"
        style={Object {}}
      >
        \`
      </span>
      <span
        className="token template-string token template-punctuation string"
        style={Object {}}
      >
        \`
      </span>
      <span
        className="token template-string token template-punctuation string"
        style={Object {}}
      >
        \`
      </span>
    </span>
  </code>
</pre>
`;
